src/hotspot/share/gc/shared/collectedHeap.hpp

```diff
@@ -435,12 +435,15 @@ class CollectedHeap : public CHeapObj<mtGC> {
   // explicitly checks if the given memory location contains a null value.
   virtual bool contains_null(const oop* p) const;
 
-  // Print heap information on the given outputStream.
-  virtual void print_on(outputStream* st) const = 0;
-  // The default behavior is to call print_on() on tty.
-  virtual void print() const;
+  // Print heap information.
+  virtual void print_heap_on(outputStream* st) const = 0;
+
+  // Print additional information about the GC that is not included in print_heap_on().
+  // Generally used for printing information in case of a fatal error.
```

Comment:
Given that this is also used in VM.info I think it would be better to skip this comment.
```suggestion
```